Integration tests fail due to Mixing RESTEasy Reactive and RESTEasy Classic client parts

### Describe the bug

After https://github.com/quarkusio/quarkus/pull/35468 if a project delares usage of `quarkus-rest-client-reactive-jackson` and uses `testcontainers-keycloak` for integration tests then Quarkus fails to start with 

`[error]: Build step io.quarkus.resteasy.reactive.common.deployment.ResteasyReactiveCommonProcessor#checkMixingStacks threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Mixing RESTEasy Reactive and RESTEasy Classic client parts is not supported`

[testcontainers-keycloak](https://github.com/dasniko/testcontainers-keycloak) happen to depend on the non-reactive `resteasy-client`.






### Expected behavior

Mixing stacks is not enforced for dependencies with the `test` scope.

### Actual behavior

Quarkus fails to start if test dependencies lead to mixing stacks.

### How to Reproduce?

1. Download the sample project - https://ro-events.s3.eu-central-1.amazonaws.com/quarkus-rest-reactive-issue.zip?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJZN3BJ2KKWHLR4WA%2F20230925%2Feu-central-1%2Fs3%2Faws4_request&X-Amz-Date=20230925T075625Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=ea0f059a8a79a76fc2c6591e7752478382c5bf2dadc39ff529eb3cecef9080b8
2. Unpack
3. `cd quarkus-rest-reactive-issue/code-with-quarkus/`
4. `mvn clean verify`


### Output of `uname -a` or `ver`

Linux pop-os 6.4.6-76060406-generic #202307241739~1694621917~22.04~ac5e1a8 SMP PREEMPT_DYNAMIC Wed S x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

OpenJDK 64-Bit Server VM Corretto-17.0.4.8.1 (build 17.0.4+8-LTS, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.4.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.4 (dfbb324ad4a7c8fb0bf182e6d91b0ae20e3d2dd9)

### Additional information

This issue may belong to the maintainers of `testcontainers-keycloak`.

Yes, the issue is still present with the version 999-SNAPSHOT:

```
[ERROR] Errors: 
[ERROR]   GreetingResourceTest.testHelloEndpoint » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.resteasy.reactive.common.deployment.ResteasyReactiveCommonProcessor#checkMixingStacks threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Mixing RESTEasy Reactive and RESTEasy Classic client parts is not supported
	at io.quarkus.resteasy.reactive.common.deployment.ResteasyReactiveCommonProcessor.checkMixingStacks(ResteasyReactiveCommonProcessor.java:105)
```

One possible workaround for Quarkus services that do not need to call Keycloak Admin APIs during tests is to exlude the KC admin client:

```xml
<dependency>
  <groupId>com.github.dasniko</groupId>
  <artifactId>testcontainers-keycloak</artifactId>
  <version>3.0.0</version>
  <scope>test</scope>
  <exclusions>
    <exclusion>
      <groupId>org.keycloak</groupId>
      <artifactId>keycloak-admin-client</artifactId>
    </exclusion>
  </exclusions>
</dependency>
```

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/deployment/src/main/java/io/quarkus/resteasy/reactive/common/deployment/ResteasyReactiveCommonProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/deployment/src/main/java/io/quarkus/resteasy/reactive/common/deployment/ResteasyReactiveCommonProcessor.java
@@ -87,17 +87,22 @@ public class ResteasyReactiveCommonProcessor {
     private static final String PROVIDERS_SERVICE_FILE = "META-INF/services/" + Providers.class.getName();
     private static final Predicate<ResolvedDependency> IS_RESTEASY_CLASSIC_CLIENT_DEP = d -> d.getArtifactId()
             .contains("resteasy-client");
+    private static final Predicate<ResolvedDependency> IS_RESTEASY_CLASSIC_CORE_DEP = d -> d.getArtifactId()
+            .equals("resteasy-core");
+    private static final Predicate<ResolvedDependency> IS_NOT_TEST_SCOPED = d -> !"test".equals(d.getScope());
 
     @Produce(ServiceStartBuildItem.class)
     @BuildStep
     void checkMixingStacks(Capabilities capabilities, CurateOutcomeBuildItem curateOutcomeBuildItem) {
         List<ResolvedDependency> resteasyClassicDeps = curateOutcomeBuildItem.getApplicationModel().getDependencies().stream()
                 .filter(d -> d.getGroupId().equals("org.jboss.resteasy")).collect(Collectors.toList());
-        boolean hasResteasyCoreDep = resteasyClassicDeps.stream().anyMatch(d -> d.getArtifactId().equals("resteasy-core"));
+        boolean hasResteasyCoreDep = resteasyClassicDeps.stream()
+                .anyMatch(IS_NOT_TEST_SCOPED.and(IS_RESTEASY_CLASSIC_CORE_DEP));
         if (!hasResteasyCoreDep) {
             return;
         }
-        boolean hasResteasyClassicClient = resteasyClassicDeps.stream().anyMatch(IS_RESTEASY_CLASSIC_CLIENT_DEP);
+        boolean hasResteasyClassicClient = resteasyClassicDeps.stream()
+                .anyMatch(IS_NOT_TEST_SCOPED.and(IS_RESTEASY_CLASSIC_CLIENT_DEP));
         if (!hasResteasyClassicClient) { // there is no bulletproof way of knowing whether a server specific dependency has been included, so we deduce it by the absence of client dependency
             throw new DeploymentException("Mixing RESTEasy Reactive and RESTEasy Classic server parts is not supported");
         }

